tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1037,6 +1038,34 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  /** Resolves any [[RecordType]] in a [[PreTransform]]. */
+  private def resolveRecordType(
+      preTrans: PreTransform): Option[(RecordType, CancelFun)] = {
```

Comment:
Consider naming this differently. It seems to be confusing w.r.t. `resolveLocalDef`. IIUC, this really just grabs the `CancelFun` of the record type, right? The type itself would already be available through `preTrans.tpe`, no?